examples/qml_features/rust/src/mock_qt_types.rs

```diff
@@ -10,18 +10,32 @@ mod ffi {
 
     unsafe extern "C++" {
         include!("cxx-qt-lib/include/qt_types.h");
+        include!("cxx-qt-lib/include/qcolor.h");
```

Comment:
Hmm, do we want to make downstreams include a header for each type? We could #include each of these in qt_types.h.

Reply:
Eventually we'll have a macro to include them all, eg `cxx_qt_lib::prelude!()` or something.

And qt_types will go away with this approach. Then we end up with smaller headers for each type you use. Which may also help later if we split up QtCore and QtGui support in cxx-qt-lib etc.

Reply:
Also the advantage of using a macro is that that can possibly also take care of all the
`type QType = cxx_qt_lib::types::QType` boilerplate.

Reply:
Yup, i'm hoping to remove the whole blocks. We could also later have like `cxx_qt_lib::qtcore::prelude!()` if we split out QtCore/QtGui to be separate in cxx-qt-lib and have separate features.

The only "fun" part is whether CXX/our macro actually causes the macro expansion to happen early enough? As if needs to happen after our macro but before the CXX one. Worst case we might need the expansion to actually happen via cxx-qt-gen ? Eg if it spots a macro called cxx_qt_lib::prelude!() then replace it for this block.

Reply:
I fear the same, I'm pretty sure the cxx_qt::bridge macro runs first, unfortunately :thinking:

Reply:
Something to battle latter, but i think the worst case of using cxx-qt-gen to do the expansion isn't terrible, We can have a tokenstream in cxx_qt_lib_headers or something that it can use, so that cxx-qt-gen doesn't depend directly on cxx-qt-lib...